cmd/gitops-bucket-server/main.go

```diff
@@ -2,60 +2,59 @@ package main
 
 import (
 	"context"
+	"flag"
 	"log"
-	"net"
 	"os"
 	"os/signal"
-	"strconv"
 	"syscall"
 
 	"github.com/johannesboyne/gofakes3"
 	"github.com/johannesboyne/gofakes3/backend/s3mem"
-	"net/http/httptest"
+	"github.com/weaveworks/weave-gitops/pkg/http"
 )
 
 func main() {
 	ctx, cancel := signal.NotifyContext(
 		context.Background(),
 		syscall.SIGINT,
-		syscall.SIGTERM,
-		syscall.SIGKILL)
+		syscall.SIGTERM)
 	defer cancel()
 
 	logger := log.New(os.Stdout, "", 0)
 	backend := s3mem.New()
 	s3 := gofakes3.New(backend,
 		gofakes3.WithAutoBucket(true),
 		gofakes3.WithLogger(gofakes3.StdLog(logger, gofakes3.LogErr, gofakes3.LogWarn, gofakes3.LogInfo)))
+	s3Server := s3.Server()
 
-	port := "9000"
-	// check args
-	if len(os.Args) > 1 {
-		port = os.Args[1]
-		// part string to integer
-		_, err := strconv.Atoi(port)
-		if err != nil {
-			log.Fatalf("Invalid port number: %s", port)
-		}
-	}
+	var (
+		httpPort, httpsPort int
+		certFile, keyFile   string
+	)
 
-	// create a listener with the desired port.
-	listener, err := net.Listen("tcp", ":"+port)
-	if err != nil {
-		log.Fatal(err)
-	}
+	flag.IntVar(&httpPort, "http-port", 9000, "TCP port to listen on for HTTP connections")
+	flag.IntVar(&httpsPort, "https-port", 9001, "TCP port to listen on for HTTPS connections")
```

Comment:
The default HTTPS port would be another number like 9443, as we are using `9001` for our default dashboard port.